internal/pkg/fieldpropagator/analyzer.go

```diff
@@ -87,6 +84,26 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return ResultType(isFieldPropagator), nil
 }
 
+func methods(ssaProg *ssa.Program, t types.Type) []*ssa.Function {
+	var methods []*ssa.Function
+	// The method sets of T and *T are disjoint.
+	// In Go code, a variable of type T has access to all
+	// the methods of type *T due to an implicit address operation.
+	methods = append(methods, methodValues(ssaProg, t)...)
+	return append(methods, methodValues(ssaProg, types.NewPointer(t))...)
+}
+
+func methodValues(ssaProg *ssa.Program, t types.Type) []*ssa.Function {
+	var methodValues []*ssa.Function
+	mset := ssaProg.MethodSets.MethodSet(t)
```

Comment:
Dos this no support a ranged for loop?

Reply:
Sadly, no. A `MethodSet` is a wrapper over a slice, but the slice isn't exposed so we have to access it through these methods.